src/lib/transaction/Transaction.svelte

```diff
@@ -83,6 +84,12 @@
                     on:click="{async () => new XBull().signTx(tx).then((signedXDR) => sendMessage(signedXDR))}"
                     >Sign Transaction with xBull</button
                 >
+            {:else if freighter}
+                <button
+                    class="simple-signer sign-tx"
+                    on:click="{async () => new Freighter().signTx(tx).then((signedXDR) => sendMessage(signedXDR))}"
+                    >Sign Transaction with Freighter</button
+                >
```

Comment:
It seems to me that these are all doing the same since they have the same interface. It'd be better to instantiate the correct wallet and assign it to a variable of interface IWallet and then do something like:

`wallet.signTx(tx).then(signedXdr => sendMessage(signedXdr))`